source/lac/scalapack.cc

```diff
@@ -323,117 +326,344 @@ void ScaLAPACKMatrix<NumberType>::invert()
 
 
 template <typename NumberType>
-std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenvalues_symmetric()
+std::vector<NumberType> ScaLAPACKMatrix<NumberType>::eigenpairs_symmetric_by_index(const std::pair<unsigned int,unsigned int> &index_limits,
+    const bool compute_eigenvectors)
+{
+  // check validity of index limits
+  Assert (index_limits.first != numbers::invalid_unsigned_int,ExcMessage("index limits have invalid value"));
+  Assert (index_limits.second != numbers::invalid_unsigned_int,ExcMessage("index limits have invalid value"));
+  Assert (index_limits.first < (unsigned int)n_rows,ExcMessage("index limits have to be smaller than matrix size"));
+  Assert (index_limits.second < (unsigned int)n_rows,ExcMessage("index limits have to be smaller than matrix size"));
+
+  std::pair<unsigned int,unsigned int> idx = std::make_pair(std::min(index_limits.first,index_limits.second),
```

Comment:
Maybe just add another assert that `index_limits.first <= index_limits.second`?

Reply:
The private function `eigenpairs_symmetric` does the correct ordering internally. This way the user does not have to worry about that detail.

Reply:
Sure, but I thought it makes more sense to ask users to provide input in the right order.

Reply:
On second glance, I did that for the following reason. If the user wants **all** eigenvalues/eigenvectors by passing the index interval [0,size-1], that would cause a call to `psyevx` instead of `psyev`, but that is not desirable.